dtos/provisionwatcher.go

```diff
@@ -19,16 +19,16 @@ type ProvisionWatcher struct {
 	Identifiers         map[string]string   `json:"identifiers" validate:"gt=0,dive,keys,required,endkeys,required"`
 	BlockingIdentifiers map[string][]string `json:"blockingIdentifiers,omitempty"`
 
-	DeviceNameTemplate string      `json:"deviceNameTemplate"`
-	ProfileName        string      `json:"profileName" validate:"omitempty,edgex-dto-rfc3986-unreserved-chars"`
+	DeviceNameTemplate string      `json:"deviceNameTemplate" validate:"omitempty,edgex-dto-none-empty-string,edgex-dto-no-reserved-chars"`
```

Comment:
the empty string is allowed for DeviceNameTemplate, ProfileName, and ProfileNameTemplate in ProvisionWatcher